### ISSUE TITLE
Error when using demo code

### ISSUE THREAD

**Issue by pittyacg** [decisionforce/TPN#10]
Hello! I am trying to run the pre-trained kinetics version on a custom video. My command is the following: 
` python ./test_video.py config_files/kinetics400/tpn/r101f32s2.py kinetics.pth --video_file video.mp4`
where kinetics.pth is the downloaded: kinetics400_tpn_r101f32s2.pth file.

I have some problems. First, I don't know exactly how should the labels file look like. The category.txt file in the demo folder does not match the Kinetics annotations.
On the other hand, the program crashes. Ar first I get this warning: 
![image](https://user-images.githubusercontent.com/10223550/83971093-09e06e80-a8e2-11ea-9200-43ff207b7638.png)
and after that it crashes with the following message:
![image](https://user-images.githubusercontent.com/10223550/83971109-1e246b80-a8e2-11ea-8635-70b87e944f78.png)

Can you help me with this problem? 

Thank you,
Petru

**Comment by limbo0000** [comment]
Hi @pittyacg,

The warning is ok since we remove the parameters of the auxiliary head which was adopted for training. Can you successfully run the `test_demo.py` with tsm_tpn trained on something-something V2?

Actually, `test_demo.py` is an example to play with our models such that it only supports 2D models (\eg TSM) trained on something-something V2 dataset. If you wanna create a file for prediction on Kinetics-400, you have to prepare the category/label txt files and change the shape of input tensor.

**Comment by pittyacg** [comment]
Thank you for your answer. I successfully run the demo on the Something-something dataset. I created the label file for Kinetics as well. Still, I am not sure where should I change the input tensor size because I did not find any code on that in the test_video.py file and I believe the configuration in the config should be fine.

Can you give me more details about that?

Thank you!

**Comment by limbo0000** [comment]
Hi @pittyacg,

Since 3D backbone takes a 5D tensor as input, you are supposed to modify the [data shape](https://github.com/decisionforce/TPN/blob/1b3568ecc0509e9210ca8f33dad98f73c8a8a80c/test_video.py#L67), refer to the [dataloader](https://github.com/decisionforce/TPN/blob/1b3568ecc0509e9210ca8f33dad98f73c8a8a80c/mmaction/datasets/rawframes_dataset.py#L304).  

I recommend fcn-testing for 3D networks to obtain the video-level prediction which averages 10*3Crop scores. [Readme under tools](https://github.com/decisionforce/TPN/tree/master/tools) provides how to test a reference model.

**Comment by limbo0000** [comment]
I would close this issue. You could reopen it if there remain some problems.
